src/stores/api.js

```diff
@@ -23,4 +27,39 @@ export async function getAllTimeGreat(lastThirtyDays=false) {
     raw: true,
   });
   return res.data;
+}
+
+export async function getLoggedInUser() {
+  try {
+    const res = await axios.get(BASE_URL + "/api/user", { raw: true });
+    const username = res.data["username"];
+    localStorage.setItem("currentUser", username);
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+export async function isUserLoggedIn() {
+  const currentUser = localStorage.getItem("currentUser");
+  if (currentUser) {
+    return currentUser;
+  }
+  return;
+}
+
+export async function LogIn() {
+  if (!localStorage.getItem("currentUser")) {
+    await getLoggedInUser();
+  }
+
+  if (!localStorage.getItem("currentUser")) {
+    window.open("http://localhost:8082/api/login");
+  }
+
+  location.reload();
+}
+
+export function LogOut() {
+  localStorage.removeItem("currentUser");
+  location.reload();
```

Comment:
We have an /api/logout endpoint that you should use as well, as it will clear the session token and everything.   I'd replace the current function with:
```
export function LogOut() {
  window.location.href = "http://localhost:8082/api/logout";
  localStorage.removeItem("currentUser");
}
```